addons/account/wizard/account_move_send.py

```diff
@@ -231,26 +231,27 @@ def _compute_send_mail_extra_fields(self):
         for wizard in self:
             wizard.enable_send_mail = wizard.mode in ('invoice_single', 'invoice_multi')
             wizard.display_mail_composer = wizard.mode == 'invoice_single'
-            send_mail_readonly = False
 
+    @api.depends('move_ids', 'checkbox_send_mail')
+    def _compute_send_mail_warning_message(self):
+        for wizard in self:
             display_messages = []
-            if wizard.enable_send_mail:
+            if wizard.mode == 'invoice_multi':
                 invoices_without_mail_data = wizard.move_ids.filtered(lambda x: not x.partner_id.email)
-                if invoices_without_mail_data:
-                    if wizard.mode == 'invoice_multi':
-                        display_messages.append(_(
-                            "The following invoice(s) will not be sent by email, because the customers don't have email "
-                            "address: "
-                        ))
-                        display_messages.append(", ".join(invoices_without_mail_data.mapped('name')))
-                        send_mail_readonly = True
-                    else:
-                        display_messages.append(_("Please add an email address for your partner"))
-
-            wizard.send_mail_readonly = send_mail_readonly
+                if invoices_without_mail_data and wizard.checkbox_send_mail:
+                    display_messages.append(_(
+                        "The partners on the following invoices have no email address, "
+                        "so those invoices will not be sent: "
+                    ))
+                    display_messages.append(", ".join(invoices_without_mail_data.mapped('name')))
             wizard.send_mail_warning_message = "".join(display_messages) if display_messages else None
 
-    @api.depends('move_ids')
+    @api.depends('move_ids', 'mail_partner_ids')
```

Comment:
Why do you add `mail_partner_ids` ?

Reply:
Because there's a double wizard that opens up when trying to send mail to partner that doesn't have email, allowing some quick edits. If we don't have the 'mail_partner_ids', we'll have to close and reopen the send wizard to see the now enabled checkbox and readonly